services/invitations/pkg/service/v0/service.go

```diff
@@ -110,7 +110,20 @@ func (s svc) Invite(ctx context.Context, invitation *invitations.Invitation) (*i
 	}
 	client := &http.Client{Transport: tr}
 
-	req, err := http.NewRequest(s.config.Endpoint.Method, urlWriter.String(), bytes.NewBufferString(bodyWriter.String()))
+	userRole := "guest"
```

Comment:
According to the MS spec the `primaryRole` is restricted to the values `student` or `teacher` IIUC. So we should probably not misuse it here.

We need to add a userType property for user at some point (see #5603) which we could set to `guest` here.

For I would be fine if we hardcode the primaryRole to `student` and at some later point add it as a property on the invitation request. Just creating "normal" non-education users would also be fine.

Reply:
> * Calling /graph/v1.0/ecuation/users with userType set as "Guest" will create a user as "Guest" and primary role as "student"

Do we actually currently require a `primaryRole` to be set? If not I think we could just leave it unset in this case.

Other than that your suggestions sound fine to me. Could you please add them to the #5603 ticket?

Reply:
@Excds I don't think we are diverging. The examples you give perfectly match what lan be found in the MS Graph docs:

The MS graph user entity *does* have a `userType` property: https://learn.microsoft.com/en-us/graph/api/resources/user?view=graph-rest-1.0#properties

Property | Type | Description
-- | -- | --
userType | String | A string value that can be used to classify user types in your directory, such as `Member` and `Guest`. Returned only on $select. Supports $filter (eq, ne, not, in, and eq on null values). NOTE: For more information about the permissions for member and guest users, see [What are the default user permissions in Azure Active Directory?](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/users-default-permissions?context=graph/context#member-and-guest-users)

The link is interesting because it explains the difference between Guest and Member users:
> * *Member users* can register applications, manage their own profile photo and mobile phone number, change their own password, and invite B2B guests. These users can also read all directory information (with a few exceptions).
> 
> * *Guest users* have restricted directory permissions. They can manage their own profile, change their own password, and retrieve some information about other users, groups, and apps. However, they can't read all directory information.
> 
> For example, guest users can't enumerate the list of all users, groups, and other directory objects. Guests can be added to administrator roles, which grant them full read and write permissions. Guests can also invite other guests.


Users created through invitations by default become `Guest` users, although company administrators can invite users as type `Member`, as described in [the list of properties for the invitation resource](https://learn.microsoft.com/en-us/graph/api/resources/invitation?view=graph-rest-1.0#properties)

Property | Type | Description
-- | -- | --
invitedUserType | String | The userType of the user being invited. By default, this is Guest. You can invite as Member if you are a company administrator.